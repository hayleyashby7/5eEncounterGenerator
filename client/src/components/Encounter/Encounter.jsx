import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { getMonstersFromAPI } from '../../utils/API';
import { isValidChallengeRating } from '../../utils/challengeRating';
import Table from '../Table/Table';

function Encounter({ challengeRating }) {
    const [monsters, setMonsters] = useState([]);
    const [error, setError] = useState(null);
    const [message, setMessage] = useState('Loading...');

    useEffect(() => {
        const fetchData = async () => {
            try {
                if (!isValidChallengeRating({ challengeRating })) {
                    throw new Error('Unable to request data.');
                }

                const data = await getMonstersFromAPI({ challengeRating });

                if (data instanceof Error) {
                    throw new Error(
                        'Unable to retrieve monsters from server. Please try again later.',
                    );
                }
                if (data.count === 0) {
                    throw new Error('No suitable monsters found.');
                }

                setMonsters(data.results);
                setError(false);
            } catch (err) {
                setError(true);
                setMonsters([]);
                setMessage(err.message);
            }
        };

        fetchData();
    }, [challengeRating]);

    return (
        <div className='bg-neutral-900 text-orange-100'>
            {error && <div>{message}</div>}{' '}
            {monsters ? (
                <Table
                    headers={[
                        { name: 'Name', data: 'name', important: true },
                        { name: 'CR', data: 'challenge_rating', important: true },
                        { name: 'Type', data: 'type', important: false },
                        { name: 'Size', data: 'size', important: false },
                        { name: 'Alignment', data: 'alignment', important: false },
                        { name: '', data: null, important: true },
                    ]}
                    data={monsters}
                    className=''
                />
            ) : (
                <p>{message}</p>
            )}{' '}
        </div>
    );
}

Encounter.propTypes = { challengeRating: PropTypes.string };
Encounter.defaultProps = { challengeRating: '-1' };

export default Encounter;
