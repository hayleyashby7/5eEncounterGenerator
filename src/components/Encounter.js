import React from 'react';

const Encounter = () => {



	return (
		<div>
			<h2>Encounter</h2>
		</div>
	);
};

export default Encounter;
