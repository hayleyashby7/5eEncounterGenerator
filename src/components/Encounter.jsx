import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Row from './Table/Row';
import { getMonstersFromAPI } from '../utils/API';
import { isValidChallengeRating } from '../utils/challengeRating';

function Encounter({ challengeRating }) {
    const [monsters, setMonsters] = useState([]);
    const [error, setError] = useState(null);
    const [message, setMessage] = useState('Loading...');

    useEffect(() => {
        const fetchData = async () => {
            try {
                if (!isValidChallengeRating({ challengeRating })) {
                    throw new Error('Unable to request data.');
                }

                const data = await getMonstersFromAPI({ challengeRating });

                if (data instanceof Error) {
                    throw new Error(
                        'Unable to retrieve monsters from server. Please try again later.',
                    );
                }
                if (data.count === 0) {
                    throw new Error('No suitable monsters found.');
                }

                setMonsters(data.results);
                setError(false);
            } catch (err) {
                setError(true);
                setMonsters([]);
                setMessage(err.message);
            }
        };

        fetchData();
    }, [challengeRating]);

    return (
        <div className='bg-neutral-900 text-orange-100'>
            <h1>List of Monsters</h1>
            {error && <div>{message}</div>}{' '}
            {monsters ? (
                <table>
                    <thead>
                        <tr>
                            <th>Name</th>
                            <th>CR</th>
                            <th>Type</th>
                            <th>Size</th>
                            <th>Alignment</th>
                        </tr>
                    </thead>
                    <tbody>
                        {monsters.map((monster) => (
                            <Row key={monster.slug} data={monster} />
                        ))}
                    </tbody>
                </table>
            ) : (
                <p>{message}</p>
            )}{' '}
        </div>
    );
}

Encounter.propTypes = { challengeRating: PropTypes.number };
Encounter.defaultProps = { challengeRating: -1 };

export default Encounter;
