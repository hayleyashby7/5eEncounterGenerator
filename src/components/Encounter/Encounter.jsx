import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { getMonstersFromAPI } from '../../utils/API';
import isValidChallengeRating from '../../utils/challengeRating';

function Encounter({ challengeRating }) {
    const [monsters, setMonsters] = useState([]);
    const [error, setError] = useState(null);
    const [message, setMessage] = useState('Loading...');

    useEffect(() => {
        const fetchData = async () => {
            try {

                if (!isValidChallengeRating({ challengeRating })) {
                    throw new Error('Unable to request data.');
                }
                
                const data = await getMonstersFromAPI({ challengeRating });

                if (data instanceof Error) {
                    throw new Error(
                        'Unable to retrieve monsters from server. Please try again later.',
                    );
                }
                if (data.count === 0) {
                    throw new Error('No suitable monsters found.');
                }

                setMonsters(data.results);
            } catch (err) {
                setError(true);
                setMessage(err.message);
            }
        };

        fetchData();
    }, [challengeRating]);

    return (
        <>
            <h1>List of Monsters</h1>
            {error && <div>{message}</div>}{' '}
            {monsters ? (
                <ul>
                    {' '}
                    {monsters.map((monster) => (
                        <li key={monster.slug}>{monster.name}</li>
                    ))}{' '}
                </ul>
            ) : (
                <p>{message}</p>
            )}{' '}
        </>
    );
}

Encounter.propTypes = { challengeRating: PropTypes.number };
Encounter.defaultProps = { challengeRating: null };

export default Encounter;
