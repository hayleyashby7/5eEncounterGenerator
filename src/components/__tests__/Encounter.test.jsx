import React from 'react';
import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import Encounter from '../Encounter';

describe('Encounter has correct header and content', () => {
	it('renders Encounter heading', () => {
		// Arrange
		const { container } = render(<Encounter />);

		// Assert
		expect(container).toMatchSnapshot();
	});
});
