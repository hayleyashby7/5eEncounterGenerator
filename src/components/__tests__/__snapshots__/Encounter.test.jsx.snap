// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Encounter has correct header and content > renders Encounter heading 1`] = `
<div>
  <div>
    <h2>
      Encounter
    </h2>
  </div>
</div>
`;
